draft-ietf-ivy-network-inventory-yang.md

```diff
@@ -626,6 +627,18 @@ Mostly, our inventory data model can cover the attributes from OpenConfig.
 
 {: numbered="false"}
 
+
+## Efficiency Issue
+
+During  the integration with OSS in some operators, some efficiency/scalability concerns have been discovered when synchronizing network inventory data for big networks.  More discussions are needed to address these concerns.
+
+Considering that relational databases are widely used by traditional OSS systems and also by some network controllers, the inventory objects are most likely to be saved in different tables. With the model defined in current draft, when doing a full synchronization, network controller needs to convert all inventory objects of each NE into component objects and combine them together into a single list, and then construct a response and send to OSS or MDSC. The OSS or MDSC needs to classify the component list and divide them into different groups, in order to save them in different tables. The combining-regrouping steps are impacting the network controller & OSS/MDSC processing, which may result in efficiency/scalability limitations in large scale networks.
+
+An alternative YANG model structure, which defines the inventory objects directly, instead of defining generic components, has also been analyzed. However, also with this model, there still could be some scalability limitations when synchronizing full inventory resources in large scale of networks. This scalability limitation is caused by the limited transmission capabilities of HTTP protocol. We think that this scalability limitation should be solved at protocol level rather than data model level.
+
+The model proposed by this draft is designed to be as generic as possible so to cover future special types of inventory objects that could be used in other technologies, that have not been identified yet. If the inventory objects were to be defined directly with fixed hierarchical relationships in YANG model, this new type of inventory objects needs to be manually defined, which is not a backward compatible change and therefore is not an acceptable approach for implementation. With a generic model, it is only needed to augment a new component class and extend some specific attributes for this new inventory component class, which is more flexible. We consider that this generic data model, enabling a flexible and backward compatible approach for other technologies, represents the main scope of this draft. Solution description to efficiency/scalability limitations mentioned above is considered as out-of-scope.
+
+
```

Comment:
In order to have an appendix number, please move this text before the `{: numbered="false"}` code

Reply:
Fixed.

Reply:
The `{: numbered="false"}` code should be kept for the Acknowledgments section which is not numbered as an appendix

I have also noted that the current md file defines an Appendix with title Appendix which is not needed (the '--- back' indicates that the following paragraphs are appendices)

Proposed changes:

OLD

--- back

```
# Appendix

## Comparison With Openconfig-platform Data Model

<...>

## Efficiency Issue

<...>

# Acknowledgments
```

NEW 

```
--- back

# Comparison With Openconfig-platform Data Model

<...>

# Efficiency Issue

<...>

{: numbered="false"}

# Acknowledgments

```

Reply:
Thanks Italo, now it is fixed.